kibit fails to load with AOT-compiled classfiles on the classpath

I have a "sanity-check" alias in my Leiningen `project.clj` file that AOT-compiles the entire application's sources as an easy way just to make sure there is no unloadable code in place.  Within the same project, then, these two commands in succession:

```
$ lein2 with-profile sanity-check do clean, compile
...
...
$ lein2 repl
```

…produces this exception:

```
Caused by: java.lang.IllegalArgumentException: No matching ctor found for class kibit.rules.util$compile_rule$fn__6921
    at clojure.lang.Reflector.invokeConstructor(Reflector.java:163)
    at clojure.lang.LispReader$EvalReader.invoke(LispReader.java:1036)
    at clojure.lang.LispReader$DispatchReader.invoke(LispReader.java:621)
    at clojure.lang.LispReader.read(LispReader.java:180)
    at clojure.lang.RT.readString(RT.java:1713)
    at kibit.rules.arithmetic__init.__init0(Unknown Source)
    at kibit.rules.arithmetic__init.<clinit>(Unknown Source)
    ... 117 more
```

A minor issue to be sure, but it's odd enough that I figured I'd file it; I don't think I've ever seen a circumstance where code won't load in the presence in AOT-compiled classfiles produced from that same code.

--- a/project.clj
+++ b/project.clj
@@ -6,8 +6,8 @@
             :distribution :repo
             :comments "Contact if any questions"}
   :dependencies [[org.clojure/clojure "1.6.0"]
-                 [org.clojure/core.logic "0.8.0-rc2"]
-                 [org.clojure/tools.cli "0.2.2"]]
-  :profiles {:dev {:dependencies [[lein-marginalia "0.7.0"]]
+                 [org.clojure/core.logic "0.8.9"]
+                 [org.clojure/tools.cli "0.3.1"]]
+  :profiles {:dev {:dependencies [[lein-marginalia "0.8.0"]]
                    :resource-paths ["test/resources"]}}
   :warn-on-reflection false)

--- a/src/kibit/check.clj
+++ b/src/kibit/check.clj
@@ -1,7 +1,7 @@
 (ns kibit.check
   "Kibit's integration point and public API"
   (:require [clojure.java.io :as io]
-            [clojure.core.logic :as logic]
+            [clojure.core.logic.unifier :as unifier]
             [kibit.core :as core]
             [kibit.rules :as core-rules]
             [kibit.reporters :as reporters])
@@ -28,7 +28,7 @@
 ;; where necessary.
 ;;
 ;; For more information, see: [rules](#kibit.rules) namespace
-(def all-rules (map logic/prep core-rules/all-rules))
+(def all-rules (map unifier/prep core-rules/all-rules))
 
 ;; Reading source files
 ;; --------------------

--- a/src/kibit/rules/util.clj
+++ b/src/kibit/rules/util.clj
@@ -1,8 +1,9 @@
 (ns kibit.rules.util
-  (:require [clojure.core.logic :as logic]))
+  (:require [clojure.core.logic :as logic]
+            [clojure.core.logic.unifier :as unifier]))
 
 (defn compile-rule [rule]
-  (let [[pat alt] (logic/prep rule)]
+  (let [[pat alt] (unifier/prep rule)]
      [(fn [expr] (logic/== expr pat))
       (fn [sbst] (logic/== sbst alt))]))
 

--- a/test/kibit/test/core.clj
+++ b/test/kibit/test/core.clj
@@ -1,11 +1,11 @@
 (ns kibit.test.core
   (:require [kibit.check :as kibit]
             [kibit.core :as core]
-            [clojure.core.logic :as logic]
+            [clojure.core.logic.unifier :as unifier]
             [kibit.rules :as core-rules])
   (:use [clojure.test]))
 
-(def all-rules  (map logic/prep core-rules/all-rules))
+(def all-rules  (map unifier/prep core-rules/all-rules))
 
 (deftest simplify-alts
   (are [expected-alt test-expr]

